### DISCUSSION TITLE
Чи потрібна Rust'у специфікація?

### DISCUSSION THREAD

**Discussion by ikalnytskyi** [shopokodu/community#15] [Теми для подкасту]
Багато гейтерів не толерують мови програмування без затрердженою ISO специфікації.

https://blog.m-ou.se/rust-standard/
https://drewdevault.com/2021/02/09/Rust-move-fast-and-break-things.html
https://blog.rust-lang.org/inside-rust/2023/11/15/spec-vision.html

План розмови:

1. Гейт від Drew DeVault для затравки:
   - C простий, Rust складний
   - Нема специфікація ==> одна реалізація, яка обмежена платформами, що підтримує LLVM і що тестує Rust Foundation
   - Треба сповільнити розвиток мови
2. Як працює розвиток мов програмування зі стандартами на прикладі C і C++
   - комітети ISO
   - компілятори від окремих вендорів
   - нові стандарти кожні декілька років через механізм пропозицій 
3. Як працює розвиток Rust:
   - єдиний офіційний компілятор
   - релізи кожні 6 тижнів
   - пропозиція фіч через RFC
   - nightly і стабілазіція фіч
   - зворотня сумісність (crater)
   - Rust Editions
   - документація
   - unsafe guidelines
4. Навіщо тоді специфікація Rust:
   - https://rust-lang.github.io/rfcs/3355-rust-spec.html
   - Ferrocene specification
   - undefined behaviour
   - сертифікація софту
   - розробка і обговорення нових RFC
   - питання щодо того, як поводяться між собою фічі Rust

**Comment by Sunday111** [comment]
Таке, на цілий подкаст не тягне. Але аспект цікавий, можна поєднати з якоюсь суміжною темою (наприклад про Mojo).
